src/redux/greetingSlice.js

```diff
@@ -0,0 +1,25 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const getGreetingMessage = createAsyncThunk('greetings/getGreetingMessage', async () => {
+  const result = await axios.get('http://localhost:3000/random_greeting');
+  return result.data;
+});
+
+const greetingSlice = createSlice({
+  name: 'greetings',
+  initialState: { value: '' },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(getGreetingMessage.fulfilled, (state, action) => {
+        if (action.payload && action.payload.content) {
+          state.value = action.payload.content;
+        } else {
+          console.error('Invalid payload structure:', action.payload);
+        }
```

Comment:
- [ ] @DevKaranJ Great work refactoring your reducer, however in the section above you are using `content` which is not in your data fetched from the backend, instead, it should be `greeting`, kindly consider checking it to fix the error of the app not displaying the greeting
![image](https://github.com/DevKaranJ/hello-react-front-end/assets/61408860/0df894b1-b513-449b-a839-70b0b7db0cd6)